cmd/opampsupervisor/e2e_test.go

```diff
@@ -0,0 +1,319 @@
+	}, 10*time.Second, 500*time.Millisecond, "Log never appeared in output")
+}
+
+func TestSupervisorRestartsCollectorAfterBadConfig(t *testing.T) {
+	var healthReport atomic.Value
+	addr, connected, send := newServer(
+		t,
+		defaultConnectingHandler,
+		server.ConnectionCallbacksStruct{
+			OnMessageFunc: func(_ types.Connection, message *protobufs.AgentToServer) *protobufs.ServerToAgent {
+				if message.Health != nil {
+					healthReport.Store(message.Health)
+				}
+
+				return &protobufs.ServerToAgent{}
+			},
+		})
+
+	s := newSupervisor(t, "basic", map[string]string{"url": addr})
+	defer s.Shutdown()
+
+	waitForConnection(connected, true)
+
+	require.Eventually(t, func() bool {
+		health := healthReport.Load().(*protobufs.AgentHealth)
+
+		if health != nil {
+			return !health.Healthy && health.LastError != ""
+		}
+
+		return false
+	}, 5*time.Second, 250*time.Millisecond, "Supervisor never reported that the Collector was unhealthy")
+
+	cfg, hash, _, _ := createSimplePipelineConfig(t)
+
+	send(&protobufs.ServerToAgent{
+		RemoteConfig: &protobufs.AgentRemoteConfig{
+			Config: &protobufs.AgentConfigMap{
+				ConfigMap: map[string]*protobufs.AgentConfigFile{
+					"": {Body: cfg.Bytes()},
+				},
+			},
+			ConfigHash: hash,
+		},
+	})
+
+	require.Eventually(t, func() bool {
+		health := healthReport.Load().(*protobufs.AgentHealth)
+
+		if health != nil {
+			return health.Healthy && health.LastError == ""
+		}
+
+		return false
+	}, 5*time.Second, 250*time.Millisecond, "Supervisor never reported that the Collector became healthy")
+}
+
+func TestSupervisorConfiguresCapabilities(t *testing.T) {
+	var capabilities atomic.Uint64
+	addr, connected, _ := newServer(
+		t,
+		defaultConnectingHandler,
+		server.ConnectionCallbacksStruct{
+			OnMessageFunc: func(_ types.Connection, message *protobufs.AgentToServer) *protobufs.ServerToAgent {
+				capabilities.Store(message.Capabilities)
+
+				return &protobufs.ServerToAgent{}
+			},
+		})
+
+	s := newSupervisor(t, "nocap", map[string]string{"url": addr})
+	defer s.Shutdown()
+
+	waitForConnection(connected, true)
+
+	require.Eventually(t, func() bool {
+		cap := capabilities.Load()
+
+		return cap == uint64(protobufs.AgentCapabilities_AgentCapabilities_ReportsStatus)
+	}, 5*time.Second, 250*time.Millisecond)
+}
+
+func createSimplePipelineConfig(t *testing.T) (*bytes.Buffer, []byte, *os.File, *os.File) {
+	wd, err := os.Getwd()
+	require.NoError(t, err)
+
+	tempDir := t.TempDir()
+	input, err := os.CreateTemp(tempDir, "input_*.yaml")
+	require.NoError(t, err)
+
+	output, err := os.CreateTemp(tempDir, "output_*.yaml")
+	require.NoError(t, err)
+
+	colCfgTpl, err := os.ReadFile(path.Join(wd, "testdata", "collector", "simple_pipeline.yaml"))
+	require.NoError(t, err)
+
+	templ, err := template.New("").Parse(string(colCfgTpl))
+	require.NoError(t, err)
+
+	var confmapBuf bytes.Buffer
+	err = templ.Execute(&confmapBuf, map[string]string{"input": input.Name(), "output": output.Name()})
```

Comment:
Is it possible to document what exactly the input/output are?

Reply:
I've updated the template parameter names to indicate that these are log files.